feat: add build functionality and tests (#12)

annelibogren opened the pull request:
Added a Build file that uses gradle to build, made two test repos, one that should compile and one that should not, and used those to test the build functionality. Used GitHandler made by Paul, " "private GitHandler(String localRepoPath, String remoteRepoURL) on line 25 of GitHandler.java must be made public for it to work.

Closes #12

annelibogren committed:
feat: add build functionality and tests (#12)

Added a Build file that uses gradle to build, made two test repos, one that should compile and one that should not, and used those to test the build functionality. Used gitHandler made by Paul.

Closes #12

annelibogren committed:
Merge branch 'main' into 12-set-up-build

paultisaw commented on src/main/java/org/example/Build.java:
```diff
@@ -0,0 +1,59 @@
+                }
+            }
+
+            process.waitFor(20, TimeUnit.SECONDS);
```
Why do we have to wait here ?

annelibogren replied:
I added a waitFor method with a timeout of 20 seconds so that it wouldn't be stuck in an infinite loop if there were issues with the execution, not sure if it is totally necessary though!

paultisaw replied:
Ok makes sense !

paultisaw commented on src/main/java/org/example/ContinuousIntegrationServer.java:
```diff
@@ -63,4 +71,30 @@ public Object handle(Request request, Response response) {
         }
     }
+
+    public static void runBuild(GitHandler gitHandler) {
```
In my part you will have the correct branch already checked out and pulled, so you don't need to worry about git commands in here
Especially since you need to build the branch to which the changes were just made, not always main

annelibogren replied:
ah ok sounds good! will fix this now!

paultisaw reviewed (commented):
It looks great, well done for making unit tests for this it must not have been easy !
If you can just explain why we wait and remove the checkout to main in your function I think we can merge it :+1:

annelibogren committed:
fix: review comments (#12)

Removed unnecessary git commands in build function in ContinuousIntegrationServer and tests

Closes #12

annelibogren committed:
Merge branch '12-set-up-build' of github.com:DD2480-Group-25/continuous-integration into 12-set-up-build

Merged to sync local and github branch

annelibogren committed:
fix: make GitHandler constructor public (#12)

Merged local branch with current git branch and made GitHandler constructor public so tests would work.

Closes #12

paultisaw reviewed (approved):
LGTM !

annelibogren committed:
fix: final tests check (#12)

Final check for testing

Closes #12
